Libraries/Opc.Ua.Server/Configuration/ConfigurationNodeManager.cs

```diff
@@ -320,10 +320,9 @@ public void HasApplicationSecureAdminAccess(ISystemContext context, string _)
                     throw new ServiceResultException(StatusCodes.BadUserAccessDenied, "Secure Application Administrator access required.");
                 }
 
-                // allow access to system configuration only through special identity
-                IUserIdentity user = context.UserIdentity as RoleBasedIdentity;               
-                if (user == null || user.TokenType == UserTokenType.Anonymous ||
-                    !user.GrantedRoleIds.Contains(ObjectIds.WellKnownRole_SecurityAdmin))
+                // allow access to system configuration only with Role SecurityAdmin
+                if (context.UserIdentity == null || context.UserIdentity.TokenType == UserTokenType.Anonymous ||
+                    !context.UserIdentity.GrantedRoleIds.Contains(ObjectIds.WellKnownRole_SecurityAdmin))
                 {
```

Comment:
please revert this change based on the previous conversation and the userIdentity helper variable for null check and further access